### PR TITLE
Cleanup XCMP `QueueConfigData`

### PR DESCRIPTION
**Pull request by serban300**
Removes obsolete fields from the `QueueConfigData` structure. For the remaining fields, if they use the old defaults, we replace them with the new defaults.

Resolves: https://github.com/paritytech/polkadot-sdk/issues/1795

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4197335

File: cumulus/pallets/xcmp-queue/src/migration.rs [hunk 1]
```diff
@@ -48,6 +50,12 @@ impl<T: Config> OnRuntimeUpgrade for MigrationToV3<T> {
 			weight.saturating_accrue(T::DbWeight::get().writes(1));
 		}
 
+		if StorageVersion::get::<Pallet<T>>() == 3 {
```
Comment (bkchr) [inline cumulus/pallets/xcmp-queue/src/migration.rs:53]: `STORAGE_VERSION` at line 30 needs to be changed to `4`.

File: cumulus/pallets/xcmp-queue/src/migration.rs [hunk 2]
```diff
@@ -48,6 +50,12 @@ impl<T: Config> OnRuntimeUpgrade for MigrationToV3<T> {
...
+			weight.saturating_accrue(migrate_to_v4::<T>());
+			StorageVersion::new(4).put::<Pallet<T>>();
+			weight.saturating_accrue(T::DbWeight::get().writes(1));
```
Comment (ggwpez) [inline cumulus/pallets/xcmp-queue/src/migration.rs:56]: The struct is called `MigrateToV3`, so i think it should not migrate to V4.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4223798

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4223797

File: cumulus/pallets/xcmp-queue/src/migration.rs (hunk 1 above)
Comment (serban300) [reply cumulus/pallets/xcmp-queue/src/migration.rs:53]: Done

File: cumulus/pallets/xcmp-queue/src/migration.rs (hunk 2 above)
Comment (serban300) [reply cumulus/pallets/xcmp-queue/src/migration.rs:56]: Changed it to `MigrateToV4`. Hope that's ok.

File: cumulus/pallets/xcmp-queue/src/migration.rs [hunk 3]
```diff
@@ -25,14 +27,14 @@ use frame_support::{
...
 
 /// The current storage version.
-pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(3);
+pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(4);
 
 pub const LOG: &str = "runtime::xcmp-queue-migration";
 
 /// Migrates the pallet storage to the most recent version.
-pub struct MigrationToV3<T: Config>(PhantomData<T>);
+pub struct MigrationToV4<T: Config>(PhantomData<T>);
```
Comment (bkchr) [inline cumulus/pallets/xcmp-queue/src/migration.rs:35]: Could you split up all these migrations into individual `MigrateToVX` structs and then also use the `VersionedMigration` from `frame_support` please.

Comment (serban300) [reply cumulus/pallets/xcmp-queue/src/migration.rs]: I didn't know about `VersionedMigration`. Done. PTAL.

File: cumulus/pallets/xcmp-queue/src/migration.rs (hunk 2 above)
Comment (serban300) [reply cumulus/pallets/xcmp-queue/src/migration.rs]: Moved each migration to a different struct as suggested in a separate comment.

File: cumulus/pallets/xcmp-queue/src/migration.rs [hunk 4]
```diff
@@ -80,6 +60,89 @@ mod v1 {
...
+
+			T::DbWeight::get().reads_writes(1, 1)
+		}
+	}
+
+	/// [`MigrationToV2`] wrapped in a
+	/// [`VersionedMigration`](frame_support::migrations::VersionedMigration), ensuring the
+	/// migration is only performed when on-chain version is 1.
+	#[allow(dead_code)]
+	pub type VersionCheckedMigrationToV2<T> = frame_support::migrations::VersionedMigration<
```
Comment (ggwpez) [inline cumulus/pallets/xcmp-queue/src/migration.rs:137]: I think the docs mention it, but i think its a bad convention to name the versioned ones with `Versioned`. Rather we should name the unchecked ones with something like `Unchecked...`. WDYT @liamaharon ?

Comment (serban300) [reply cumulus/pallets/xcmp-queue/src/migration.rs]: Sounds good. Done.

File: cumulus/pallets/xcmp-queue/src/migration.rs [hunk 5]
```diff
@@ -121,95 +188,125 @@ pub mod v3 {
...
+	pub struct UncheckedMigrationToV4<T: Config>(PhantomData<T>);
+
+	impl<T: Config> OnRuntimeUpgrade for UncheckedMigrationToV4<T> {
+		fn on_runtime_upgrade() -> Weight {
+			let translate = |pre: v2::QueueConfigData| -> QueueConfigData {
+				use sp_std::cmp::max;
+
+				let default = QueueConfigData::default();
+				let post = QueueConfigData {
+					suspend_threshold: max(pre.suspend_threshold, default.suspend_threshold),
```
Comment (ggwpez) [inline cumulus/pallets/xcmp-queue/src/migration.rs:281]: Why `max` here?

File: cumulus/pallets/xcmp-queue/src/migration.rs [hunk 6]
```diff
@@ -247,4 +347,64 @@ mod tests {
...
+				resume_threshold: 1,
+				..Default::default()
+			};
+
+			frame_support::storage::unhashed::put_raw(
+				&crate::QueueConfig::<Test>::hashed_key(),
+				&v2.encode(),
+			);
+
+			v4::MigrationToV4::<Test>::on_runtime_upgrade();
```
Comment (ggwpez) [inline cumulus/pallets/xcmp-queue/src/migration.rs:370]: You can also call the pre and post hooks.

File: cumulus/parachains/runtimes/contracts/contracts-rococo/src/lib.rs [hunk 7]
```diff
@@ -100,7 +100,7 @@ pub type UncheckedExtrinsic =
 /// Migrations to apply on runtime upgrade.
 pub type Migrations = (
 	cumulus_pallet_parachain_system::migration::Migration<Runtime>,
-	cumulus_pallet_xcmp_queue::migration::MigrationToV3<Runtime>,
```
Comment (ggwpez) [inline cumulus/parachains/runtimes/contracts/contracts-rococo/src/lib.rs:103]: I think we either have to keep all of them in, or remove all.  
And please always put new migrations at the end.

**Review by ggwpez — APPROVED** [review]
Looks broadly good, thanks!

File: cumulus/pallets/xcmp-queue/src/migration.rs (hunk 6 above)
Comment (serban300) [reply cumulus/pallets/xcmp-queue/src/migration.rs:377]: Done

File: cumulus/parachains/runtimes/contracts/contracts-rococo/src/lib.rs (hunk 7 above)
Comment (serban300) [reply cumulus/parachains/runtimes/contracts/contracts-rococo/src/lib.rs:103]: Done

File: cumulus/pallets/xcmp-queue/src/migration.rs (hunk 5 above)
Comment (serban300) [reply cumulus/pallets/xcmp-queue/src/migration.rs]: Changed the strategy to something safer. If the values are the old defaults, migrate them to the new defaults. If not, keep them as they are. WDYT ?

Comment (ggwpez) [reply cumulus/pallets/xcmp-queue/src/migration.rs]: Ah okay. Since these are all thresholds they are mostly safe to be increased 👍

**Review by acatangiu — APPROVED** [review]

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v1-5-0/5358/1

**Pull request merged**